controllers/storagecluster/cephcluster.go

```diff
@@ -1184,6 +1200,44 @@ func (r *StorageClusterReconciler) setReadAffinity(sc *ocsv1.StorageCluster) err
 	return nil
 }
 
+func (r *StorageClusterReconciler) setCephFSKernelMountOption(sc *ocsv1.StorageCluster) error {
+	var cephFSKernelMountOption = "ms_mode=prefer-crc"
+	var restartNeeded = false
+	// If In-transit encryption is enabled set cephFsKernelMountOption to "ms_mode=secure"
+	if sc.Spec.Network != nil && sc.Spec.Network.Connections != nil && sc.Spec.Network.Connections.Encryption != nil && sc.Spec.Network.Connections.Encryption.Enabled {
+		cephFSKernelMountOption = "ms_mode=secure"
+	}
+	cm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      util.OcsOperatorConfigName,
+			Namespace: sc.Namespace,
+		},
+		Data: map[string]string{
+			"CSI_CEPHFS_KERNEL_MOUNT_OPTIONS": cephFSKernelMountOption,
+		},
+	}
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, cm, func() error {
```

Comment:
instead of `restartNeeded` you can make use of the result returned from `CreateOrUpdate` itself.